pkg/network/ovn_kubernetes.go

```diff
@@ -813,45 +822,53 @@ func validateOVNKubernetes(conf *operv1.NetworkSpec) []error {
 		if oc.GenevePort != nil && (*oc.GenevePort < 1 || *oc.GenevePort > 65535) {
 			out = append(out, errors.Errorf("invalid GenevePort %d", *oc.GenevePort))
 		}
-		var v4Net, v6Net *net.IPNet
+		var v4JoinNet, v6JoinNet, v4MasqNet, v6MasqNet *net.IPNet
 		var err error
 		if oc.V4InternalSubnet != "" {
 			if !cnHasIPv4 {
-				out = append(out, errors.Errorf("v4InternalSubnet and ClusterNetwork must have matching IP families"))
+				out = append(out, errors.Errorf("v4InternalSubnet %s and ClusterNetwork must have matching IP families", oc.V4InternalSubnet))
 			}
-			_, v4Net, err = net.ParseCIDR(oc.V4InternalSubnet)
+			_, v4JoinNet, err = net.ParseCIDR(oc.V4InternalSubnet)
 			if err != nil {
 				out = append(out, errors.Errorf("v4InternalSubnet is invalid: %s", err))
 			}
+			if v4JoinNet == nil {
+				out = append(out, errors.Errorf("Unable to parse cidr for v4InternalSubnet %s", oc.V4InternalSubnet))
+				return out
+			}
```

Comment:
If `v4JoinNet == nil` then `err != nil` so it doesn't make sense to check and append an error for both things.
Also this method is trying to validate all without breaking out early, I think we can keep doing that.

I would do this
```
_, v4JoinNet, err = net.ParseCIDR(oc.V4InternalSubnet)
if err != nil {
    out = append(out, errors.Errorf("v4InternalSubnet is invalid: %s", err))
} else if !isV4InternalSubnetLargeEnough(conf) {
    out = append(out, errors.Errorf("v4InternalSubnet %s is not large enough for the maximum number of nodes which can be supported by ClusterNetwork", oc.V4InternalSubnet))
}

...
...

    if v4JoinNet != nil {
        _, v4ClusterNet, _ := net.ParseCIDR(cn.CIDR)
	if iputil.NetsOverlap(*v4JoinNet, *v4ClusterNet) {
	    out = append(out, errors.Errorf("v4InternalSubnet %s overlaps with ClusterNetwork %s", oc.V4InternalSubnet, cn.CIDR))
	}
    }
```

Similar approach on the other instances.

Reply:
latest change implements this

Reply:
You missed the second part, I added a comment for it